core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -639,19 +640,22 @@ private[spark] object Utils extends Logging {
       fs: FileSystem,
       conf: SparkConf,
       hadoopConf: Configuration,
-      fileOverwrite: Boolean): Unit = {
-    if (!targetDir.mkdir()) {
+      fileOverwrite: Boolean,
+      filename: Option[String] = None): Unit = {
```

Comment:
can we add a comment that says this is only expected to be set if the path describes a file, not a directory?